`generate_nodejs.go` generates code with references to deprecated `pulumi.utils.liftProperties`

Due to a recent change in the Pulumi SDK

https://github.com/pulumi/pulumi/commit/66bd3f4aa8f9a90d3de667828dda4bed6e115f6b#diff-69f6fda7a76725a360a3a56363853252L109

The function `pulumi.utils.liftProperties` no longer exists (it had been commented as deprecated earlier)

However, `tfgen/generate_nodejs.go` will still generate code that includes references to it.

https://github.com/pulumi/pulumi-terraform-bridge/blob/a82bc5eadceca3d207599f6cb3febe8e6f44fa1f/pkg/tfgen/generate_nodejs.go#L1112

Generating a NodeJS terraform bridge and attempting to use it will fail due to these references

```
getEscalationPolicy.ts:44:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

44     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getExtensionSchema.ts:63:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

63     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getSchedule.ts:49:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

49     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getService.ts:27:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

27     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getTeam.ts:53:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

53     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getUser.ts:49:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

49     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~

getVendor.ts:63:25 - error TS2339: Property 'liftProperties' does not exist on type 'typeof import("/Users/bharbron/git/pulumi-pagerduty/sdk/nodejs/node_modules/@pulumi/pulumi/utils")'.

63     return pulumi.utils.liftProperties(promise, opts);
                           ~~~~~~~~~~~~~~


Found 7 errors.
```

--- a/pkg/tfbridge/info.go
+++ b/pkg/tfbridge/info.go
@@ -249,7 +249,6 @@ type JavaScriptInfo struct {
 	PeerDependencies  map[string]string // NPM peer-dependencies to add to package.json.
 	Overlay           *OverlayInfo      // optional overlay information for augmented code-generation.
 	TypeScriptVersion string            // A specific version of TypeScript to include in package.json.
-	AsyncDataSources  bool              // If true, only generate the async version of data-sources
 }
 
 // PythonInfo contains optional overlay information for Python code-generation.

--- a/pkg/tfgen/generate_nodejs.go
+++ b/pkg/tfgen/generate_nodejs.go
@@ -1068,13 +1068,8 @@ func (g *nodeJSGenerator) emitResourceFunc(mod *module, fun *resourceFunc, neste
 		retty = fun.retst.name
 	}
 
-	if g.info.JavaScript.AsyncDataSources {
-		w.Writefmtln("export function %s(%sopts?: pulumi.InvokeOptions): Promise<%s> {",
-			fun.name, argsig, retty)
-	} else {
-		w.Writefmtln("export function %s(%sopts?: pulumi.InvokeOptions): Promise<%s> & %s {",
-			fun.name, argsig, retty, retty)
-	}
+	w.Writefmtln("export function %s(%sopts?: pulumi.InvokeOptions): Promise<%s> {",
+		fun.name, argsig, retty)
 
 	if fun.info.DeprecationMessage != "" {
 		w.Writefmtln("    pulumi.log.warn(\"%s is deprecated: %s\")", fun.name, fun.info.DeprecationMessage)
@@ -1094,23 +1089,14 @@ func (g *nodeJSGenerator) emitResourceFunc(mod *module, fun *resourceFunc, neste
 	w.Writefmtln("        opts.version = utilities.getVersion();")
 	w.Writefmtln("    }")
 
-	if g.info.JavaScript.AsyncDataSources {
-		// Now simply invoke the runtime function with the arguments, returning the results.
-		w.Writefmtln("    return pulumi.runtime.invoke(\"%s\", {", fun.info.Tok)
-	} else {
-		w.Writefmtln("    const promise: Promise<%s> = pulumi.runtime.invoke(\"%s\", {", retty, fun.info.Tok)
-	}
+	// Now simply invoke the runtime function with the arguments, returning the results.
+	w.Writefmtln("    return pulumi.runtime.invoke(\"%s\", {", fun.info.Tok)
 
 	for _, arg := range fun.args {
 		// Pass the argument to the invocation.
 		w.Writefmtln("        \"%[1]s\": args.%[1]s,", arg.name)
 	}
 	w.Writefmtln("    }, opts);")
-
-	if !g.info.JavaScript.AsyncDataSources {
-		w.Writefmtln("")
-		w.Writefmtln("    return pulumi.utils.liftProperties(promise, opts);")
-	}
 
 	w.Writefmtln("}")
 

